dvol_python/test_dvol.py

```diff
@@ -134,6 +140,65 @@ def test_create_volume_with_path_separator(self):
         self.assertIn("Error", output)
         self.assertIn("foo/bar", output)
 
+    def _parse_list_output(self, dvol):
+        """
+        Return a tuple containing the first line of output, and then the
+        remaining lines of output from the last command output in the dvol
+        object.
+        """
+        lines = dvol.voluminous.getOutput()[-1].split("\n")
+        return lines[0].split(), [line.split() for line in lines[1:]]
+
+    def test_switch_active_volume(self):
+        """
+        ``dvol switch`` should switch the currently active volume displayed
+        with a ``*`` in ``dvol list`` output.
+
+        Assert both blackboxy and whiteboxy things about the implementation,
+        because we care about upgradeability (wrt on-disk format) between
+        different implementations.
```

Comment:
This paragraph is false now, since you've taken out the `list`-based (whiteboxy) assertions. Change it to: "Assert whiteboxy things about the implementation, because we care about upgradeability (wrt on-disk format) between different implementations."